packages/framework/src/Component/FileUpload/FileUpload.php

```diff
@@ -266,13 +270,21 @@ protected function getPositionForNewEntity(EntityFileUploadInterface $entity): i
             return $this->positionByEntityAndType[$entityName][$entityId][$uploadEntityType][$type];
         }
 
-        $position = $this->imageRepository->getImagesCountByEntityIndexedById(
-            $entityName,
-            $entityId,
-            $type,
-        );
-
-        $this->positionByEntityAndType[$entityName][$entityId][$uploadEntityType][$type] = $position;
+        $position = match ($uploadEntityType) {
+            'image' => $this->imageRepository->getImagesCountByEntityIndexedById(
+                $entityName,
+                $entityId,
+                $type,
+            ),
+            'customerFile' => $this->customerUploadedFileRepository->getCustomerUploadedFilesCountByEntityIndexedById(
```

Comment:
In some cases, it is not correct. If a file is deleted, the count is lower than the highest position.